notebook/notebookapp.py

```diff
@@ -1759,8 +1768,8 @@ def init_terminals(self):
         try:
             from .terminal import initialize
             initialize(nb_app=self)
-            self.web_app.settings['terminals_available'] = True
         except ImportError as e:
+            self.terminals_enabled = False
```

Comment:
Modifying this still seems like it's inviting confusion, because what `.terminals_enabled` means depends on the point at which you use it: before this, it means a config option, and after this it means whether the terminal machinery is used.

Can I suggest that we keep clearly separate terms for the three things we might want to know, e.g. `terminals_available`, `terminals_enabled` and `terminals_in_use`?

Reply:
+1, although I believe `terminado_available` as an indication of the package's presence is more clear than `terminals_available` if that's okay.

I also think it's important to issue the warning message when terminals are enabled but the terminado is not present even though we know that answer prior to `init_terminals`.  I've retained that try-catch in case there might be other initialization failures down the road.

Reply:
No problem - I wasn't tied to those specific names, just keen to distinguish the concepts.

Now `terminals_available` on the webapp is the equivalent of `terminals_in_use` on the Jupyter app class. It's possibly worth aligning those, but I don't think it's a big deal either.